common/cache/domainCache.go

```diff
@@ -952,3 +958,14 @@ func (entry *DomainCacheEntry) IsSampledForLongerRetention(
 	}
 	return false
 }
+
+func (entry *DomainCacheEntry) HoursToAddToRetentionDuration() time.Duration {
```

Comment:
we need to write tests for this method, test cases:
 * no key -> 0
 * not a number -> 0
 * a number -> number